packages/ckeditor5-ui/tests/colorpicker/colorpickerview.js

```diff
@@ -50,31 +50,179 @@ describe( 'ColorPickerView', () => {
 		} );
 
 		it( 'should update color property after getting an input', () => {
-			view.input.fieldView.value = '#ff0000';
-			view.input.fieldView.fire( 'input' );
+			view.inputRow.children.get( 1 ).fieldView.value = '#ff0000';
+			view.inputRow.children.get( 1 ).fieldView.fire( 'input' );
 
 			clock.tick( 200 );
 
 			expect( view.color ).to.equal( '#ff0000' );
 		} );
 
-		// Requires https://github.com/ckeditor/ckeditor5/pull/13819 - text input should be properly registered as a focusable.
-		it( 'should not update color property during input editing when focused', () => {
-			view.color = '#000000';
+		describe( 'should update color property', () => {
+			describe( 'when hex color was passed', () => {
+				describe( 'in 3-character format', () => {
+					it( 'with `#` at the beginning', () => {
+						testColorUpdateFromInput( {
+							inputValue: '#abc',
+							expectedInput: '#abc',
+							expectedColorProperty: '#abc'
+						} );
+					} );
+
+					it( 'without `#` at the beginning', () => {
+						testColorUpdateFromInput( {
+							inputValue: 'abc',
+							expectedInput: 'abc',
+							expectedColorProperty: '#abc'
+						} );
+					} );
+				} );
+
+				describe( 'in 4-character format', () => {
+					it( 'with `#` at the beginning', () => {
+						testColorUpdateFromInput( {
+							inputValue: '#abcd',
+							expectedInput: '#abcd',
+							expectedColorProperty: '#abcd'
+						} );
+					} );
+
+					it( 'without `#` at the beginning', () => {
+						testColorUpdateFromInput( {
+							inputValue: 'abcd',
+							expectedInput: 'abcd',
+							expectedColorProperty: '#abcd'
+						} );
+					} );
+				} );
+
+				describe( 'in 6-character format', () => {
+					it( 'with `#` at the beginning', () => {
+						testColorUpdateFromInput( {
+							inputValue: '#abcdef',
+							expectedInput: '#abcdef',
+							expectedColorProperty: '#abcdef'
+						} );
+					} );
+
+					it( 'without `#` at the beginning', () => {
+						testColorUpdateFromInput( {
+							inputValue: 'abcdef',
+							expectedInput: 'abcdef',
+							expectedColorProperty: '#abcdef'
+						} );
+					} );
+				} );
+
+				describe( 'in 8-character format', () => {
+					it( 'with `#` at the beginning', () => {
+						testColorUpdateFromInput( {
+							inputValue: '#abcdefab',
+							expectedInput: '#abcdefab',
+							expectedColorProperty: '#abcdefab'
+						} );
+					} );
+
+					it( 'without `#` at the beginning', () => {
+						testColorUpdateFromInput( {
+							inputValue: 'abcdefab',
+							expectedInput: 'abcdefab',
+							expectedColorProperty: '#abcdefab'
+						} );
+					} );
+				} );
+			} );
+		} );
 
-			view.input.isFocused = true;
-			view.input.fieldView.value = '#ffffff';
-			view.input.fieldView.fire( 'input' );
+		describe( 'should not update color property', () => {
+			// Requires https://github.com/ckeditor/ckeditor5/pull/13819 - text input should be properly registered as a focusable.
```

Comment:
Since we're at it this line should be removed before - when removing `.skip()` on this test 🙄 let's fix it while we're at it